admin/classes/domain/SushiService.php

```diff
@@ -599,91 +630,177 @@ private function parseXML($fName, $reportLayout, $overwritePlatform){
 				$name = $reader->getAttribute("Name");
 				$version = $reader->getAttribute("Version");
 
-				$layoutCode = $name;	
+				$layoutCode = $name;
 
 				if (($version == "3") || ($version =="4")){
 					$version = "R" . $version;
 				}
 				if ($version != ''){
-					$layoutCode .= "_" . $version;		
+					$layoutCode .= "_" . $version;
 				}else{
-					$layoutCode .= "_R" . $this->releaseNumber; 
+					$layoutCode .= "_R" . $this->releaseNumber;
 				}
 
 				//At this point, determine the format of the report to port to csv from the layouts.ini file
 				$layoutKey = $layoutsArray['ReportTypes'][$layoutCode];
-	  			$layoutColumns = $layoutsArray[$layoutKey]['columns'];
+				$layoutColumns = $layoutsArray[$layoutKey]['columns'];
 
 
-	  			//if this way of determining layout was unsuccessful, just use the layout sent in
-	  			if (count($layoutColumns) == "0"){
-	  				$layoutCode = $reportLayout . "_R" . $this->releaseNumber;
+				//if this way of determining layout was unsuccessful, just use the layout sent in
+				if (count($layoutColumns) == "0"){
+					$layoutCode = $reportLayout . "_R" . $this->releaseNumber;
 
 					$layoutKey = $layoutsArray['ReportTypes'][$layoutCode];
-		  			$layoutColumns = $layoutsArray[$layoutKey]['columns'];
-	  			}
-				
+					$layoutColumns = $layoutsArray[$layoutKey]['columns'];
+
+					///////////////////////////////////////////////////////
+					// Create header for SUSHI file
+					///////////////////////////////////////////////////////
+					$header = $layoutColumns;
+					for ($i = 0; $i < sizeof($header); $i++) {
+						if ($header[$i] == "jan" && $startMonth == 1) {
```

Comment:
Can we condense this logic to something like the following?

$startMonthArray = array('jan' => 1, 'feb' => 2, ... 'dec' => 12);
foreach ($startMonthArray as $monthName => $monthNumber) {
  if ($header[$i] == $monthName && $startMonth <= $monthNumber) {
    $header[$i] .= "-$startYear";
  } else {
    $header[$i] .= "-$endYear";
  }
}